experiments/ssl_experiments/pimodel_mcdropout_cifar10.py

```diff
@@ -3,25 +3,35 @@
 from argparse import Namespace
 
 import torch
-from baal.active import ActiveLearningDataset, get_heuristic
-from baal.bayesian.dropout import patch_module
-from baal.utils.pytorch_lightning import ActiveLearningMixin, BaalTrainer, ResetCallback
+from pimodel_cifar10 import PIModel
 from torch import nn
+from torch.hub import load_state_dict_from_url
 from torch.utils.data import DataLoader
 from torchvision.datasets import CIFAR10
 from torchvision.models import vgg16
 
-from experiments.ssl_experiments.pimodel_cifar10 import PIModel
+from baal.active import ActiveLearningDataset, get_heuristic
+from baal.bayesian.dropout import patch_module
+from baal.utils.pytorch_lightning import ActiveLearningMixin, BaalTrainer, ResetCallback
 
 
 class PIActiveLearningModel(ActiveLearningMixin, PIModel):
-    def __init__(self, active_dataset: ActiveLearningDataset, hparams: Namespace, network: nn.Module):
+    def __init__(self, active_dataset: ActiveLearningDataset, hparams: Namespace,
+                 network: nn.Module):
         super().__init__(active_dataset, hparams, network)
 
         self.network = patch_module(self.network)
 
     def pool_loader(self):
-        return DataLoader(self.active_dataset.pool, self.hparams.batch_size, shuffle=False)
+        return DataLoader(self.active_dataset.pool, self.hparams.batch_size, shuffle=False,
+                          num_workers=4)
+
+    def configure_optimizers(self):
+        return torch.optim.SGD(self.parameters(), lr=self.hparams.lr, weight_decay=1e-4)
+
+    def optimizer_step(self, epoch_nb, batch_nb, optimizer, optimizer_i, opt_closure, **kwargs):
```

Comment:
Why do we override this method?

Reply:
Adam is very bad on small datasets. Could be configurable

Reply:
isn't this the default behavior for optimizer_step though ?

Reply:
You do not want the lr rampup for the active learning ?

Reply:
Not really, in AL we overfit so quickly in the beginning that lr schedules are not usable/useful.
In anycase, do you think we could replace PIModel rampup with something from torch.optim.lr_scheduler and not override this method?

Reply:
yes I did it this way to avoid duplicating code because the rampup is used for the unsupervised weight also

Reply:
I could also test the pi model without the lr rampup if we don't mind not replicating the paper exactly

Reply:
yeah let'S try that to keep the example clean.:)